### PR TITLE
fby3.5: rf: Support PM8702 sensor reading

### PR DESCRIPTION
**Pull request by quanta-Irenelin**
Summary:
- Initial the MCTP instance on RF BIC
- Set Endpoint ID for pm8702 via MCTP CCI
- Support to read the chip temperature of PM8702
- Support to read the temperature of the DIMMs

Test Plan:
Build code: Pass
```
root@bmc-oob:~# sensor-util slot1 | grep RF
RF MB_Inlet Temp             (0x50) :   35.00 C     | (ok)
RF CXL Temp                  (0x51) :   43.00 C     | (ok)
RF CXL_Cntr Temp             (0x52) :   37.00 C     | (ok)
RF DIMM_A Temp               (0x53) :   34.75 C     | (ok)
RF DIMM_B Temp               (0x54) :   35.00 C     | (ok)
RF DIMM_C Temp               (0x55) :   34.25 C     | (ok)
RF DIMM_D Temp               (0x57) :   33.25 C     | (ok)
```

Signed-off-by: Irene <Irene.Lin@quantatw.com>

### CODE REVIEW

File: meta-facebook/yv35-rf/src/platform/plat_mctp.c [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+		if (bus == p->conf.smbus_conf.bus) {
+			return p->mctp_inst;
+		}
+	}
+	return NULL;
+}
+
+static void set_endpoint_resp_handler(void *args, uint8_t *buf, uint16_t len)
+{
+	if (!buf || !len)
```
Comment (garnermic) [inline meta-facebook/yv35-rf/src/platform/plat_mctp.c:65]: Please use CHECK_NULL_ARG for testing parameters against null.

File: meta-facebook/yv35-rf/src/platform/plat_mctp.c [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+		return;
+	LOG_HEXDUMP_INF(buf, len, __func__);
+}
+
+static void set_endpoint_resp_timeout(void *args)
+{
+	mctp_route_entry *p = (mctp_route_entry *)args;
```
Comment (garnermic) [inline meta-facebook/yv35-rf/src/platform/plat_mctp.c:72]: Please add a CHECK_NULL_ARG to ensure args is not NULL.

File: meta-facebook/yv35-rf/src/platform/plat_mctp.c [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+	printk("[%s] Endpoint 0x%x set endpoint failed on bus %d \n", __func__, p->endpoint,
+	       p->bus);
+}
+
+static void set_dev_endpoint(void)
+{
+	for (uint8_t i = 0; i < ARRAY_SIZE(mctp_route_tbl); i++) {
+		mctp_route_entry *p = mctp_route_tbl + i;
+		for (uint8_t j = 0; j < ARRAY_SIZE(smbus_port); j++) {
```
Comment (garnermic) [inline meta-facebook/yv35-rf/src/platform/plat_mctp.c:81]: Please check to see *p is not NULL before dereferencing in line 82.

File: meta-facebook/yv35-rf/src/platform/plat_mctp.c [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+			msg.timeout_cb_fn_args = p;
+
+			mctp_ctrl_send_msg(find_mctp_by_smbus(p->bus), &msg);
+		}
+	}
+}
+
+static uint8_t mctp_msg_recv(void *mctp_p, uint8_t *buf, uint32_t len, mctp_ext_params ext_params)
+{
+	if (!mctp_p || !buf || !len)
```
Comment (garnermic) [inline meta-facebook/yv35-rf/src/platform/plat_mctp.c:112]: Please use CHECK_NULL_ARG_WITH_RETURN to test for NULL values.

Also, is len always expected to be zero or is it supposed to be a pointer to len (e.g. uint32_t *len)?

File: meta-facebook/yv35-rf/src/platform/plat_mctp.c [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+		LOG_WRN("Cannot find message receive function!!");
+		return MCTP_ERROR;
+	}
+
+	return MCTP_SUCCESS;
+}
+
+uint8_t get_mctp_route_info(uint8_t dest_endpoint, void **mctp_inst, mctp_ext_params *ext_params)
+{
+	if (!mctp_inst || !ext_params)
```
Comment (garnermic) [inline meta-facebook/yv35-rf/src/platform/plat_mctp.c:137]: Please use CHECK_NULL_ARG_WITH_RETURN macro for NULL checks where a value is returned.

File: meta-facebook/yv35-rf/src/platform/plat_mctp.c [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+			rc = MCTP_SUCCESS;
+			break;
+		}
+	}
+	return rc;
+}
+
+uint8_t get_mctp_info(uint8_t dest_endpoint, mctp **mctp_inst, mctp_ext_params *ext_params)
+{
+	if (!mctp_inst || !ext_params) {
```
Comment (garnermic) [inline meta-facebook/yv35-rf/src/platform/plat_mctp.c:158]: Please use CHECK_NULL_ARG_WITH_RETURN

File: meta-facebook/yv35-rf/src/platform/plat_mctp.c [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+	k_work_submit(&send_cmd_work);
+}
+
+void plat_mctp_init()
+{
+	LOG_INF("plat_mctp_init");
+	for (uint8_t i = 0; i < ARRAY_SIZE(smbus_port); i++) {
+		mctp_smbus_port *p = smbus_port + i;
+		/* init the mctp/cci instance */
+		// mctp_smbus_port *p = smbus_port;
```
Comment (garnermic) [inline meta-facebook/yv35-rf/src/platform/plat_mctp.c:193]: Please remove dead code.

File: meta-facebook/yv35-rf/src/platform/plat_mctp.c [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+void send_cmd_to_dev(struct k_timer *timer)
+{
+	k_work_submit(&send_cmd_work);
+}
+
+void plat_mctp_init()
+{
+	LOG_INF("plat_mctp_init");
+	for (uint8_t i = 0; i < ARRAY_SIZE(smbus_port); i++) {
+		mctp_smbus_port *p = smbus_port + i;
```
Comment (garnermic) [inline meta-facebook/yv35-rf/src/platform/plat_mctp.c:191]: Please check *p is not NULL before using.

File: meta-facebook/yv35-rf/src/platform/plat_sdr_table.c [hunk 9]
```diff
@@ -22,6 +22,311 @@
...
+		0x00, // LNRT
+		0x00, // LCT
+		0x00, // LNCT
+		0x00, // positive-going threshold
+		0x00, // negative-going threshold
+		0x00, // reserved
+		0x00, // reserved
+		0x00, // OEM
+		IPMI_SDR_STRING_TYPE_ASCII_8, // ID len, should be same as "size of struct"
+		"RF CXL Temp",
```
Comment (garnermic) [inline meta-facebook/yv35-rf/src/platform/plat_sdr_table.c:84]: These sensor names do not conform to the sensor naming standard.

File: meta-facebook/yv35-rf/src/platform/plat_sdr_table.c [hunk 10]
```diff
@@ -22,6 +22,311 @@
...
+		0x00, // LNRT
+		0x00, // LCT
+		0x00, // LNCT
+		0x00, // positive-going threshold
+		0x00, // negative-going threshold
+		0x00, // reserved
+		0x00, // reserved
+		0x00, // OEM
+		IPMI_SDR_STRING_TYPE_ASCII_8, // ID len, should be same as "size of struct"
+		"RF DIMM_A Temp",
```
Comment (garnermic) [inline meta-facebook/yv35-rf/src/platform/plat_sdr_table.c:145]: These sensor names do not conform to the sensor naming standard.

Same for all sensors below.

**Review by garnermic — CHANGES_REQUESTED** [review]

File: meta-facebook/yv35-rf/src/platform/plat_mctp.c [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+static mctp_route_entry mctp_route_tbl[] = {
+	{ CXL_EID, I2C_BUS_CXL, I2C_ADDR_CXL0 },
+};
+
+mctp *find_mctp_by_smbus(uint8_t bus)
+{
+	uint8_t i;
+	for (i = 0; i < ARRAY_SIZE(smbus_port); i++) {
+		mctp_smbus_port *p = smbus_port + i;
+		if (bus == p->conf.smbus_conf.bus) {
```
Comment (garnermic) [inline meta-facebook/yv35-rf/src/platform/plat_mctp.c:57]: Please add check to ensure p is not NULL before dereferencing.

**Review by garnermic — CHANGES_REQUESTED** [review]
Couple of instances where null pointer checks still needed.

File: meta-facebook/yv35-rf/src/platform/plat_sdr_table.c (hunk 10 above)
Comment (quanta-Irenelin) [reply meta-facebook/yv35-rf/src/platform/plat_sdr_table.c]: > These sensor names do not conform to the sensor naming standard.
> 
> Same for all sensors below.

Thank you for your suggestion; I already modified the sensor name in the #d17bb89 commit of PR #829.
Thanks.

```
root@bmc-oob:~# sensor-util slot1 | grep RF
RF MB_Inlet Temp             (0x50) :   35.00 C     | (ok)
RF_CXL_TEMP_C                (0x51) :   43.00 C     | (ok)
RF CXL_Cntr Temp             (0x52) :   36.00 C     | (ok)
RF_DIMM_A_TEMP_C             (0x53) :   34.50 C     | (ok)
RF_DIMM_B_TEMP_C             (0x54) :   34.75 C     | (ok)
RF_DIMM_C_TEMP_C             (0x55) :   34.25 C     | (ok)
RF_DIMM_D_TEMP_C             (0x57) :   33.00 C     | (ok)

```

**Comment by quanta-Irenelin** [comment]
Force-push this branch to fix a couple of pointer-check issues.
I have added the check-null function and checked the pointer in the #752d1c1 commit of PR #829.
Would you please review it again?
Many thanks!

**Pull request closed without merging**